src/components/AccountDrawer/MiniPortfolio/PortfolioLogo.tsx

```diff
@@ -87,81 +78,114 @@ const L2LogoContainer = styled.div<{ $backgroundColor?: string }>`
   justify-content: center;
 `
 
-/**
- * Renders an image by prioritizing a list of sources, and then eventually a fallback triangle alert
- */
-export function PortfolioLogo({
-  chainId = ChainId.MAINNET,
-  accountAddress,
-  currencies,
-  images,
-  size = '40px',
-  style,
-}: MultiLogoProps) {
-  const chainInfo = getChainInfo(chainId)
-  const squareLogoUrl = chainInfo?.squareLogoUrl
-  const logoUrl = chainInfo?.logoUrl
-  const chainLogo = squareLogoUrl ?? logoUrl
+type DoubleLogoProps = {
+  logo1?: string
+  logo2?: string
+  size: string
+  onError1?: () => void
+  onError2?: () => void
+}
+
+function DoubleLogo({ logo1, onError1, logo2, onError2, size }: DoubleLogoProps) {
+  return (
+    <DoubleLogoContainer>
+      <CircleLogoImage size={size} src={logo1 ?? blankTokenUrl} onError={onError1} />
+      <CircleLogoImage size={size} src={logo2 ?? blankTokenUrl} onError={onError2} />
+    </DoubleLogoContainer>
+  )
+}
+
+type DoubleCurrencyLogoProps = {
+  chainId: ChainId
+  currencies: Array<Currency | undefined>
+  backupImages?: (string | undefined)[]
+  size: string
+}
+
+function DoubleCurrencyLogo({ chainId, currencies, backupImages, size }: DoubleCurrencyLogoProps) {
+  const [src, nextSrc] = useTokenLogoSource(
+    currencies?.[0]?.wrapped.address,
+    chainId,
+    currencies?.[0]?.isNative,
+    backupImages?.[0]
+  )
+  const [src2, nextSrc2] = useTokenLogoSource(
+    currencies?.[1]?.wrapped.address,
+    chainId,
+    currencies?.[1]?.isNative,
+    backupImages?.[1]
+  )
+
+  return currencies.length > 1 ? (
+    <DoubleLogo logo1={src} onError1={nextSrc} logo2={src2} onError2={nextSrc2} size={size} />
+  ) : currencies.length === 1 && src ? (
+    <CircleLogoImage size={size} src={src} onError={nextSrc} />
+  ) : (
+    <MissingImageLogo size={size}>
+      {currencies[0]?.symbol?.toUpperCase().replace('$', '').replace(/\s+/g, '').slice(0, 3)}
+    </MissingImageLogo>
+  )
+}
```

Comment:
nit: slightly hard to decipher initially maybe break it down like this - same styling suggestion for the code below

```  
  if (currencies.length === 1 && src) {
    return CircleLogoImage size={size} src={src} onError={nextSrc} />
  }
  
  if (currencies.length > 1) {
    return <DoubleLogo logo1={src} onError1={nextSrc} logo2={src2} onError2={nextSrc2} size={size} />
  } 
  
  return(
    <MissingImageLogo size={size}>
      {currencies[0]?.symbol?.toUpperCase().replace('$', '').replace(/\s+/g, '').slice(0, 3)}
    </MissingImageLogo>
  )
```